src/Sulu/Bundle/ContentBundle/Markup/Link/PageLinkProvider.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\ContentBundle\Markup\Link;
+
+use Sulu\Component\Content\Repository\Content;
+use Sulu\Component\Content\Repository\ContentRepositoryInterface;
+use Sulu\Component\Content\Repository\Mapping\MappingBuilder;
+use Sulu\Component\Webspace\Manager\WebspaceManagerInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Integrates pages into link-system.
+ */
+class PageLinkProvider implements LinkProviderInterface
```

Comment:
is tested functional with `Sulu\Bundle\ContentBundle\Tests\Functional\Markup\LinkTagTest`